ntoskrnl/mm/ARM3/pagfault.c

```diff
@@ -2401,31 +2219,15 @@ _WARN("Session space stuff is not implemented yet!")
         ProtoPte = MiCheckVirtualAddress(Address, &ProtectionCode, &Vad);
         if (ProtectionCode == MM_NOACCESS)
         {
-#if (_MI_PAGING_LEVELS == 2)
-            /* Could be a page table for paged pool */
-            MiCheckPdeForPagedPool(Address);
-#endif
-            /* Has the code above changed anything -- is this now a valid PTE? */
-            Status = (PointerPte->u.Hard.Valid == 1) ? STATUS_SUCCESS : STATUS_ACCESS_VIOLATION;
-
-            /* Either this was a bogus VA or we've fixed up a paged pool PDE */
-            if (TrapInformation)
-                MiUnlockProcessWorkingSet(CurrentProcess, CurrentThread);
-            return Status;
+            /* This was a bogus VA */
+            return STATUS_ACCESS_VIOLATION;
         }
 
         /*
          * Check if this is a real user-mode address or actually a kernel-mode
          * page table for a user mode address
          */
-        if (Address <= MM_HIGHEST_USER_ADDRESS
-#if _MI_PAGING_LEVELS >= 3
-            || MiIsUserPte(Address)
-#if _MI_PAGING_LEVELS == 4
-            || MiIsUserPde(Address)
-#endif
-#endif
-        )
+        if (Address <= MM_HIGHEST_USER_ADDRESS)
```

Comment:
Can't this be asserted? I don't see how this could be a page table address here now.

Reply:
I'm unclear whether a page fault on hyper space address could occur for now. But this was checkd above, so I guess we can assert it.